services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks/_core.py

```diff
@@ -265,3 +267,19 @@ async def delete_tasks_from_project(self, project_id: ProjectID) -> None:
             await conn.execute(
                 sa.delete(comp_tasks).where(comp_tasks.c.project_id == f"{project_id}")
             )
+
+    async def get_outputs_from_tasks(
+        self, project_id: ProjectID, node_ids: set[NodeID]
+    ) -> dict[NodeID, dict[str, Any]]:
+        selection = list(map(str, node_ids))
+        query = sa.select(comp_tasks.c.node_id, comp_tasks.c.outputs).where(
+            (comp_tasks.c.project_id == f"{project_id}")
+            & (comp_tasks.c.node_id.in_(selection))
+        )
+        async with self.db_engine.acquire() as conn:
+            result: ResultProxy = await conn.execute(query)
+            rows: list[RowProxy] | None = await result.fetchall()
+            if rows:
+                assert set(selection) == {f"{_.node_id}" for _ in rows}  # nosec
```

Comment:
you double check that SQL makes it right??? that is offensive programming no ? ;)

Reply:
you mean defensive :-) I hope I am not "offending" anybody here :-)
Yes, this is kind of a way to double check what I was expecting. It is true that is redundant but since it does not run in production it does not really matter